client/my-sites/plans/controller.jsx

```diff
@@ -74,18 +75,17 @@ module.exports = {
 		analytics.tracks.recordEvent( 'calypso_plans_view' );
 		analytics.pageView.record( analyticsBasePath, analyticsPageTitle );
 
-		ReactDom.render(
-			<ReduxProvider store={ context.store }>
-				<CartData>
-					<Plans
-						sites={ sites }
-						onSelectPlan={ handlePlanSelect }
-						plans={ plans }
-						context={ context }
-						destinationType={ context.params.destinationType }/>
-				</CartData>
-			</ReduxProvider>,
-			document.getElementById( 'primary' )
+		renderWithReduxStore(
+			<CartData>
+				<Plans
+					site={ site }
```

Comment:
This method is only called when the `/plans/:site` route is hit, so when a user switches from site `foo` to site `bar`, this method is called and `Plans` receives new props. 

However, when the user visits this route and local storage is empty, `site` is not available until the sites are fetched, and because the route doesn't change once the sites are fetched, a new `site` prop is never provided to `Plans`. This results in a [perpetual loading state](https://cloudup.com/cNaiWh_xJmn), even once the sites have loaded.